packages/common/src/registry.ts

```diff
@@ -0,0 +1,115 @@
+import ProjectRegistryABI from "./abis/ProjectRegistry.json";
+import { ethers } from "ethers";
+import { Provider } from "@wagmi/core";
+import { Web3Provider } from "@ethersproject/providers";
+import {
+  Multicall,
+  ContractCallContext,
+} from "ethereum-multicall";
+
+export const fetchProjectOwners = async (
+  provider: Provider | Web3Provider,
+  chainID: number,
+  projectID: number,
+): Promise<string[]> => {
+  const addresses = addressesByChainID(chainID);
+
+  const projectRegistry = new ethers.Contract(
+    addresses.projectRegistry!,
+    ProjectRegistryABI,
+    provider,
+  );
+
+  try {
+    const projectOwners = await projectRegistry.getProjectOwners(projectID);
+    return projectOwners as string[];
+  } catch (err) {
+    // console.log(err);
+    console.log(
+      `Error fetching owners from registry on chain ${chainID} for project ${projectID}`,
+    );
+    return [];
+  }
+};
+
+export const fetchMultipleProjectOwners = async (
+  provider: Provider | Web3Provider,
+  chainID: number,
+  projectIDs: number[],
+): Promise<string[][]> => {
+
+  const addresses = addressesByChainID(chainID);
+  const registryAddress = addresses.projectRegistry!;
+
+  const multicall = new Multicall({
+    ethersProvider: provider,
+    tryAggregate: true,
+  });
+
+  const contractCallContext: ContractCallContext[] = [
+    {
+      reference: "projectRegistry",
+      contractAddress: registryAddress,
+      abi: ProjectRegistryABI,
+      calls: [],
+    },
+  ]
+
+  for (const projectID of projectIDs) {
+    contractCallContext[0].calls.push({
```

Comment:
If we want to batch multiple calls in one, JSON-RPC already supports this natively, it allows you to make multiple calls and get their results in one single call, instead of introducing a third party library and contract.

Reply:
I would like to update this but should we do in another PR so this is not held up any longer @boudra ?

Reply:
Which also begs the question that if we're not in a rush to deploy this anymore we should think about using the indexer to get the applications, it should simplify this PR a lot.